lochness/box/__init__.py

```diff
@@ -68,15 +82,104 @@ def get_access_token(client_id: str, client_secret: str, user_id: str) -> str:
     data = {"client_id": client_id,
             "client_secret": client_secret,
             "grant_type": "client_credentials",
-            "box_subject_type": "user",
-            "box_subject_id": user_id}
+            "box_subject_type": "enterprise",
+            "box_subject_id": enterprise_id}
+
+    response = requests.post(url, headers=headers, data=data)
+
+    try:
+        api_token = response.json()['access_token']
+    except KeyError:
+        raise TokenAccessError(
+                'Error in getting token using client credentials')
+
+    return api_token
+
+
+def refresh_access_token(
+        refresh_token_path: Path, client_id: str, client_secret: str) -> str:
+    '''Refresh access token using Box API
+
+    Key Argument:
+        Lochness: refresh_token_path
+        client_id: Client ID from the box app from box dev console, str
+        client_secret: Client secret from the box app from box dev console, str
+        refresh_token: Refresh token, str.
+
+    Returns:
+        api_token: API TOKEN used to pull data, str.
+
+    Notes:
+        To obtain Refresh token, go to 
+        https://account.box.com/ \
+            api/oauth2/authorize?client_id=CLIENTID&response_type=code
+        using a web browser. Once logged in, the address returned on the
+        web-browser contains code.
+
+        Then the curl command below could be used to obtain initial access
+        and refresh token.
+        curl --location --request POST 'https://api.box.com/oauth2/token' \
+            --header 'Content-Type: application/x-www-form-urlencoded' \
+            --data-urlencode 'client_id=CLIENTID' \
+            --data-urlencode 'client_secret=CLIENTSECRET' \
+            --data-urlencode 'grant_type=authorization_code' \
+            --data-urlencode 'code=CODE'
+
+        The refresh token expires when used to refresh access token. 
+        Refreshing token returns a new access and refresh token.
+        
+    '''
```

Comment:
In the whole docstring, just one piece of information is missing--why do I need to obtain a new refresh token? Is it because that token automatically expires every X minutes preventing future download?

Reply:
Correct. Just for your information, since Pronet's box app has been authorized by Yale Box, this function will not be used.

Commit coming soon.
```
    When the box app in use is not authorized by your institution, getting an
    access token with client ID, secret and enterprise_id will not work. This
    function is a workaround for the situation. For this workaround, you need
    a refresh token, which you can obtain by following instruction in the
    notes, and saving it to `.refresh_token` file, in the same directory as
    your keyring file.
```

Reply:
c6c350c